### PR TITLE
Fix #192 warnings for scipy mad func and other deprecated params

### LINKED ISSUE #192: Fix warnings
Fix these these warnings that pop up upon testing

```
pytest
======================================================================== test session starts ========================================================================
platform darwin -- Python 3.9.6, pytest-7.1.1, pluggy-1.0.0
rootdir: /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer
collected 129 items                                                                                                                                                 

pycytominer/tests/test_aggregate.py .........                                                                                                                 [  6%]
pycytominer/tests/test_annotate.py ...                                                                                                                        [  9%]
pycytominer/tests/test_consensus.py ..                                                                                                                        [ 10%]
pycytominer/tests/test_feature_select.py .........                                                                                                            [ 17%]
pycytominer/tests/test_normalize.py ...........                                                                                                               [ 26%]
pycytominer/tests/test_cyto_utils/test_DeepProfiler_processing.py .....                                                                                       [ 30%]
pycytominer/tests/test_cyto_utils/test_annotate_custom.py ......                                                                                              [ 34%]
pycytominer/tests/test_cyto_utils/test_cells.py .................                                                                                             [ 48%]
pycytominer/tests/test_cyto_utils/test_cp_image_features.py ..                                                                                                [ 49%]
pycytominer/tests/test_cyto_utils/test_feature_blocklist.py ..                                                                                                [ 51%]
pycytominer/tests/test_cyto_utils/test_feature_drop_outlier.py ....                                                                                           [ 54%]
pycytominer/tests/test_cyto_utils/test_feature_infer.py .....                                                                                                 [ 58%]
pycytominer/tests/test_cyto_utils/test_features_count_na.py .                                                                                                 [ 58%]
pycytominer/tests/test_cyto_utils/test_features_util.py .                                                                                                     [ 59%]
pycytominer/tests/test_cyto_utils/test_load.py ....                                                                                                           [ 62%]
pycytominer/tests/test_cyto_utils/test_modz.py .......                                                                                                        [ 68%]
pycytominer/tests/test_cyto_utils/test_output.py ......                                                                                                       [ 72%]
pycytominer/tests/test_cyto_utils/test_single_cell_ingest_utils.py ...                                                                                        [ 75%]
pycytominer/tests/test_cyto_utils/test_util.py ............                                                                                                   [ 84%]
pycytominer/tests/test_cyto_utils/test_write_gct.py ....                                                                                                      [ 87%]
pycytominer/tests/test_operations/test_correlation_threshold.py ....                                                                                          [ 90%]
pycytominer/tests/test_operations/test_get_na_columns.py ...                                                                                                  [ 93%]
pycytominer/tests/test_operations/test_sparse_random_projection.py ..                                                                                         [ 94%]
pycytominer/tests/test_operations/test_transform.py ....                                                                                                      [ 97%]
pycytominer/tests/test_operations/test_variance_threshold.py ...                                                                                              [100%]

========================================================================= warnings summary ==========================================================================
pycytominer/cyto_utils/util.py:61
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/cyto_utils/util.py:61: UserWarning: Non-canonical compartment detected: new
    warnings.warn(warn_str)

pycytominer/tests/test_consensus.py: 6 warnings
pycytominer/tests/test_feature_select.py: 5 warnings
pycytominer/tests/test_cyto_utils/test_modz.py: 20 warnings
pycytominer/tests/test_cyto_utils/test_util.py: 1 warning
pycytominer/tests/test_operations/test_correlation_threshold.py: 5 warnings
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/cyto_utils/util.py:338: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.tril(np.ones(data_cor_natri_df.shape), k=-1).astype(np.bool)

pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
pycytominer/tests/test_feature_select.py::test_feature_select_noise_removal
  /usr/local/lib/python3.9/site-packages/numpy/core/fromnumeric.py:3613: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
    return std(axis=axis, dtype=dtype, out=out, ddof=ddof, **kwargs)

pycytominer/tests/test_normalize.py::test_normalize_robustize_mad_allsamples
pycytominer/tests/test_normalize.py::test_normalize_robustize_mad_allsamples_novar
pycytominer/tests/test_operations/test_transform.py::test_robust_mad
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/operations/transform.py:131: DeprecationWarning: `median_absolute_deviation` is deprecated, use `median_abs_deviation` instead!
  
  To preserve the existing default behavior, use
  `scipy.stats.median_abs_deviation(..., scale=1/1.4826)`.
  The value 1.4826 is not numerically precise for scaling
  with a normal distribution. For a numerically precise value, use
  `scipy.stats.median_abs_deviation(..., scale='normal')`.
  
    median_absolute_deviation(X, nan_policy="omit"), index=self.median.index

pycytominer/tests/test_cyto_utils/test_cells.py::test_merge_single_cells
pycytominer/tests/test_cyto_utils/test_cells.py::test_merge_single_cells
pycytominer/tests/test_cyto_utils/test_cells.py::test_merge_single_cells
pycytominer/tests/test_cyto_utils/test_cells.py::test_merge_single_cells
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/cyto_utils/cells.py:693: FutureWarning: Passing 'suffixes' which cause duplicate columns {'ObjectNumber_cytoplasm'} in the result is deprecated and will raise a MergeError in a future version.
    sc_df = sc_df.merge(

pycytominer/tests/test_cyto_utils/test_modz.py::test_modz_base
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_modz.py:48: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_series_equal(

pycytominer/tests/test_cyto_utils/test_modz.py::test_modz
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_modz.py:71: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_frame_equal(

pycytominer/tests/test_cyto_utils/test_modz.py::test_modz_multiple_columns
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_modz.py:118: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_frame_equal(

pycytominer/tests/test_cyto_utils/test_modz.py::test_modz_multiple_columns_one_metadata_column
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_modz.py:148: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_frame_equal(

pycytominer/tests/test_cyto_utils/test_output.py::test_compress
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_output.py:57: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_frame_equal(

pycytominer/tests/test_cyto_utils/test_output.py::test_compress_tsv
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_output.py:74: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_frame_equal(

pycytominer/tests/test_cyto_utils/test_output.py::test_output_none
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_cyto_utils/test_output.py:90: FutureWarning: The 'check_less_precise' keyword in testing.assert_*_equal is deprecated and will be removed in a future version. You can stop passing 'check_less_precise' to silence this warning.
    pd.testing.assert_frame_equal(

pycytominer/tests/test_operations/test_transform.py::test_low_variance_spherize
  /usr/local/lib/python3.9/site-packages/numpy/lib/function_base.py:2829: RuntimeWarning: invalid value encountered in true_divide
    c /= stddev[:, None]

pycytominer/tests/test_operations/test_transform.py::test_low_variance_spherize
  /usr/local/lib/python3.9/site-packages/numpy/lib/function_base.py:2830: RuntimeWarning: invalid value encountered in true_divide
    c /= stddev[None, :]

pycytominer/tests/test_operations/test_transform.py::test_robust_mad
  /Users/shsingh/work/repos/software/cytominer/cytomining/pycytominer/pycytominer/tests/test_operations/test_transform.py:86: DeprecationWarning: `median_absolute_deviation` is deprecated, use `median_abs_deviation` instead!
  
  To preserve the existing default behavior, use
  `scipy.stats.median_abs_deviation(..., scale=1/1.4826)`.
  The value 1.4826 is not numerically precise for scaling
  with a normal distribution. For a numerically precise value, use
  `scipy.stats.median_abs_deviation(..., scale='normal')`.
  
    result = median_absolute_deviation(transform_df).sum()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================ 129 passed, 62 warnings in 63.04s (0:01:03) ============================================================
(pycytominer) pycytominer$ 
```

### PR DESCRIPTION
**Pull request by johnarevalo**
# Description

Change deprecated methods from external libraries.

## What is the nature of your change?

- [x] Bug fix (fixes an issue).

# Checklist

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: pycytominer/operations/transform.py
Comment (johnarevalo) [inline pycytominer/operations/transform.py:173]: The `scale` param is required to preserve previous behavior. More info at [scipy doc](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.median_absolute_deviation.html#scipy.stats.median_absolute_deviation)

**Comment by johnarevalo** [comment]
@gwaybio . This PR is ready. 5 warnings still persist. Two are related to `np.corrcoef` in constant columns, expected (?).

I found the remainder three harder to fix. This method is generating  duplicated column names when `pd.merge` with suffixes.

https://github.com/cytomining/pycytominer/blob/b4d32d39534c949ad5165f0b98b79537c2a7ca58/pycytominer/cyto_utils/cells.py#L604

**Review by gwaybio — CHANGES_REQUESTED** [review]
One inline comment.

The merge error is interesting. In your digging, did you notice any more details about what was going on? Is it safe to ignore this warning (pasted below)? (It's the first time I'm seeing it, to be honest)

```bash
 pycytominer/tests/test_cyto_utils/test_cells.py::test_merge_single_cells
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/pandas/core/frame.py:9203: FutureWarning: Passing 'suffixes' which cause duplicate columns {'ObjectNumber_cytoplasm'} in the result is deprecated and will raise a MergeError in a future version.
    validate=validate,
```

Given that 5 errors still persist, and @shntnu opened #192, I think he should have a say in how we proceed.

A couple other notes that might help our decision:

- We're moving away from the `SingleCells().merge_single_cells()` functionality, and solving this future error might not be necessary (unless the warning is pointing to some logic error! 😬 )
- We're also moving away from sqlalchemy, and same comment about these other 3 errors.
- My vote for how to proceed is to merge after addressing comments, but Shantanu, please chime in if you have any other concerns.

**Comment by johnarevalo** [comment]
Thanks for this review Greg.

> did you notice any more details about what was going on?

IIUC `merge_single_cells` method expects certain naming convention generated by pandas to address duplicate column names after merging two DataFrames (e.g. add `_a` and `_b` suffixes). Future pandas will not resolve duplicate columns issues, but will raise an error instead.

> We're moving away from the SingleCells().merge_single_cells() functionality, and solving this future error might not be necessary (unless the warning is pointing to some logic error! 😬 )

I'd expect this too. Removing such method would solve this warning.

**Comment by gwaybio** [comment]
Sorry for my delay in getting back to this (it dropped off the radar, to be honest!). Off the radar until today when @axiomcura noted an error in his pycytominer testing (and [pycytominer's requirements for scipy =>=1.3](https://github.com/cytomining/pycytominer/blob/b4d32d39534c949ad5165f0b98b79537c2a7ca58/requirements.txt#L2)). Thanks for your detective work Erik!

It seems that this PR will fix this error, and keep our scipy requirements up-to-date

It seems to me that we're all set to merge this in after Shantanu's 👍 :

> > My vote for how to proceed is to merge after addressing comments, but Shantanu, please chime in if you have any other concerns.

> I like this plan

----
p.s. - I opened #217 to capture Shantanu's sleuthing in https://github.com/cytomining/pycytominer/pull/194#issuecomment-1180781387

**Review by gwaybio — APPROVED** [review]

**Pull request merged**
